### DISCUSSION TITLE
Shopware Migration to Cloud Storage

### DISCUSSION THREAD

**Discussion by thesec** [FriendsOfShopware/FroshPlatformThumbnailProcessor#68] [Q&A] [answered]
I would like to migrate SW6 from local file storage to cloud storage and plan to use your thumbnail processor (kudos!) with it.

The question I can find no answer for in the docs at https://developer.shopware.com/docs/guides/hosting/infrastructure/filesystem#amazon-s3 or the SW forum is ...

How to migrate the existing original (image) files?
Do I have to manually replace each of them to trigger the upload to the cloud storage?
Is there a script doing the upload to any of the usual suspects (like S3 or Google Cloud)?

Thanks in advance :)

**Comment by tinect** [comment, marked as answer]
Hey, thank you for writing.

Your request isn't suitable to this plugin and repo as it doesn't save or transfer any file.

Please reach the shopware support.
Try community Slack f.e.: https://slack.shopware.com/

**Comment by thesec** [comment]
Hey,

You advertise this plugin to handle files on cloud storage and provide the configurations for the named ones ... the migration being a prerequisite to use your plugin with it ... but I understand you wanting to keep the discussions with plugin-only q&a.

Thanks @tinect.

**Reply by tinect** [reply]
Hey @thesec 

Oh, there is a misunderstanding, sorry for that. This plugin will not work as intended with default cloud storages. It works just with services (and a few cloud providers) who provides dynamic image processing. Nothing more. But as far as I know, amazon s3 or google don't

Could you please help me find out, where I advertise cloud storage for this plugin? For me, this is clear, but definitely I could use wrong words somewhere.

**Reply by thesec** [reply]
Ups, now you find me confused :/

It's been the plugin page in the store that I interpreted for it providing cloud storage / CDN support to images:

https://store.shopware.com/en/frosh69611263569f/thumbnailprocessor-incl.-lazy-loading-and-webp-support.html
```{mediaUrl}: Primarily your config shopware.cdn.url, alternatively shop url```

and the patterns linked to directly below that at
https://github.com/FriendsOfShopware/FroshPlatformThumbnailProcessor/discussions/categories/patterns
which include e.g. Cloudimage and bunnyCDN.

The misunderstanding / where I started to climb a different tree is when looking into Cloudimage itself and the options it provides as file serving sources.

**Reply by tinect** [reply]
Hey @thesec,

thank you for the feedback.
Please keep in mind, that CDN (Content Delivery Network) and cloud storage are different topics and things.
Sure, a CDN can offer cloud storage. 

But basically a CDN can serve files by picking from any server (also your shops url).

I hope this helps :-)
